pmd-core/src/main/java/net/sourceforge/pmd/properties/builders/PropertyDescriptorBuilder.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.properties.builders;
+
+import org.apache.commons.lang3.StringUtils;
+
+import net.sourceforge.pmd.properties.PropertyDescriptor;
+
+
+/**
+ * Base class for property builders.
+ *
+ * @param <E> Value type of the built descriptor
+ * @param <T> Concrete type of this builder instance. Removes code duplication at the expense of a few unchecked casts.
+ *            Everything goes well if this parameter's value is correctly set.
+ *
+ * @author Clément Fournier
+ * @since 6.0.0
+ */
+public abstract class PropertyDescriptorBuilder<E, T extends PropertyDescriptorBuilder<E, T>> {
+
+    protected String name;
```

Comment:
considering the name is required (there is no way to use a property without a name), I'd suggest having a constructor to set it directly instead of having to call `name()` each time, which will:

1. reduce errors
2. remove some boilerplate

Reply:
I'm guessing, it replaces the boilerplate by some other boilerplate ^^ But I agree with the rationale